parsing/parser.mly

```diff
@@ -2498,8 +2498,8 @@ with_type_binder:
 /* Polymorphic types */
 
 typevar_list:
-        QUOTE mkrhs(ident)                      { [$2] }
-      | typevar_list QUOTE mkrhs(ident)         { $3 :: $1 }
+        mkrhs(QUOTEIDENT)                      { [$1] }
+      | typevar_list mkrhs(QUOTEIDENT)         { $2 :: $1 }
```

Comment:
Note that this changes the location attributed by `mkrhs` (probably in a good direction).